lib/Jsrt/ChakraCommon.h

```diff
@@ -2978,4 +2978,343 @@ typedef unsigned char* ChakraBytePtr;
+    /// <param name="write">If the handle was opened for writing.</param>
+    typedef void (CALLBACK *JsTTDFlushAndCloseStreamCallback)(_In_ HANDLE strm, _In_ bool read, _In_ bool write);
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Creates a new runtime in Record Mode.
+    /// </summary>
+    /// <param name="attributes">The attributes of the runtime to be created.</param>
+    /// <param name="infoUri">The uri where the recorded Time-Travel data should be stored.</param>
+    /// <param name="snapInterval">The interval to wait between snapshots (measured in millis).</param>
+    /// <param name="snapHistoryLength">The length of time to maintain before discadring (measured in number of snapshots).</param>
+    /// <param name="threadService">The thread service for the runtime. Can be null.</param>
+    /// <param name="runtime">The runtime created.</param>
+    /// <remarks>See <c>JsCreateRuntime</c> for additional information.</remarks>
+    /// <returns>
+    ///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+    /// </returns>
+    CHAKRA_API
+        JsTTDCreateRecordRuntime(
+            _In_ JsRuntimeAttributes attributes,
+            _In_z_ wchar_t* infoUri,
+            _In_ UINT32 snapInterval, 
+            _In_ UINT32 snapHistoryLength,
+            _In_opt_ JsThreadServiceCallback threadService,
+            _Out_ JsRuntimeHandle *runtime);
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Creates a new runtime in Debug Mode.
+    /// </summary>
+    /// <param name="attributes">The attributes of the runtime to be created.</param>
+    /// <param name="infoUri">The uri where the recorded Time-Travel data should be loaded from.</param>
+    /// <param name="threadService">The thread service for the runtime. Can be null.</param>
+    /// <param name="runtime">The runtime created.</param>
+    /// <remarks>See <c>JsCreateRuntime</c> for additional information.</remarks>
+    /// <returns>
+    ///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+    /// </returns>
+    CHAKRA_API
+        JsTTDCreateDebugRuntime(
+            _In_ JsRuntimeAttributes attributes,
+            _In_z_ wchar_t* infoUri,
+            _In_opt_ JsThreadServiceCallback threadService,
+            _Out_ JsRuntimeHandle *runtime);
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Creates a script context in Time Travel mode for running scripts.
+    /// </summary>
+    /// <remarks>See <c>JsCreateContext</c> for more information.</remarks>
+    /// <param name="runtime">The runtime the script context is being created in.</param>
+    /// <param name="newContext">The created script context.</param>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API
+        JsTTDCreateContext(
```

Comment:
Can we call JsCreateContext and set it as current context on a runtime created using JsTTDCreateDebugRuntime?